osf/utils/notifications.py

```diff
@@ -0,0 +1,98 @@
+from django.utils import timezone
+from website.mails import mails
+from website.reviews import signals as reviews_signals
+from website.settings import DOMAIN, OSF_SUPPORT_EMAIL, OSF_CONTACT_EMAIL
+
+
+def get_email_template_context(resource):
+    from osf.models import Preprint, Registration
+    assert isinstance(resource, (Registration, Preprint)), 'This resource does not fit the template'
+    url_segment = 'preprints' if isinstance(resource, Preprint) else 'registries'
+    return {
+        'domain': DOMAIN,
+        'reviewable': resource,
+        'workflow': resource.provider.reviews_workflow,
+        'provider_url': resource.provider.domain or f'{DOMAIN}{url_segment}/{resource.provider._id}',
+        'provider_contact_email': resource.provider.email_contact or OSF_CONTACT_EMAIL,
+        'provider_support_email': resource.provider.email_support or OSF_SUPPORT_EMAIL,
+        'document_type': getattr(resource, 'preprint_word', 'registration')
+    }
+
+
+def notify_submit(resource, user, *args, **kwargs):
+    context = get_email_template_context(resource)
+    context['referrer'] = user
+    recipients = list(resource.contributors)
+    reviews_signals.reviews_email_submit.send(
+        context=context,
+        recipients=recipients
+    )
+    reviews_signals.reviews_email_submit_moderators_notifications.send(
+        timestamp=timezone.now(),
+        context=context
+    )
+
+
+def notify_resubmit(resource, user, action, *args, **kwargs):
+    context = get_email_template_context(resource)
+    reviews_signals.reviews_email.send(
+        creator=user,
+        context=context,
+        template='reviews_resubmission_confirmation',
+        action=action
+    )
+
+
+def notify_accept_reject(resource, user, action, states, *args, **kwargs):
```

Comment:
It would be somewhat significant we'd have to break up the digests and transnational emails, it's unclear how long that would take as we re-using a lot of preprint machine code.

Reply:
sounds good